File: Leetcode/BinaryTreePostorderTraversal.py
# Given the root of a binary tree, return the postorder traversal of its nodes' values.

class Solution(object):
    def postorderTraversal(self, root):
        if not root:
            return []
        
        return self.postorderTraversal(root.left) + self.postorderTraversal(root.right) + [root.val]

File: Leetcode/easy/BinaryTreePostorderTraversal.scala
// Given the root of a binary tree, return the postorder traversal of its nodes' values.
// Time O(n), Space O(n)

object Solution {
    def postorderTraversal(root: TreeNode): List[Int] = {
        if (root == null) List()
        else postorderTraversal(root.left) ::: postorderTraversal(root.right) ::: List(root.value)
    }
}